src/main/java/no/fint/p360/data/kulturminne/TilskuddFartoyFactory.java

```diff
@@ -51,31 +50,34 @@ public TilskuddFartoyResource toFintResource(Case caseResult) throws GetDocument
 
         TilskuddFartoyResource tilskuddFartoy = new TilskuddFartoyResource();
         tilskuddFartoy.setSoknadsnummer(FintUtils.createIdentifikator(caseResult.getExternalId().getId()));
-        return noarkFactory.getSaksmappe(caseDefaults.getTilskuddfartoy(), caseResult, tilskuddFartoy);
-
-        /*
-        String caseNumber = caseResult.getCaseNumber();
-        String caseYear = NOARKUtils.getCaseYear(caseNumber);
-        String sequenceNumber = NOARKUtils.getCaseSequenceNumber(caseNumber);
-        tilskuddFartoy.addSelf(Link.with(TilskuddFartoy.class, "mappeid", caseYear, sequenceNumber));
-        tilskuddFartoy.addSelf(Link.with(TilskuddFartoy.class, "systemid", caseResult.getRecno().toString()));
-        tilskuddFartoy.addSelf(Link.with(TilskuddFartoy.class, "soknadsnummer", caseResult.getExternalId().getId()));
-         */
 
+        return noarkFactory.getSaksmappe(properties, caseResult, tilskuddFartoy);
     }
 
-
     public CreateCaseArgs convertToCreateCase(TilskuddFartoyResource tilskuddFartoy) {
-        CreateCaseArgs createCaseArgs = noarkFactory.createCaseArgs(caseDefaults.getTilskuddfartoy(), tilskuddFartoy);
+        CreateCaseArgs createCaseArgs = noarkFactory.createCaseArgs(properties, tilskuddFartoy);
         createCaseArgs.setExternalId(P360Utils.getExternalIdParameter(tilskuddFartoy.getSoknadsnummer()));
         if (StringUtils.isNotBlank(project)) {
             createCaseArgs.setProject(project);
         }
+
+        if (contextUser != null && StringUtils.isNotBlank(contextUser.getUsername())) {
+            createCaseArgs.setADContextUser(contextUser.getUsername());
+            log.info("CreateCaseArgs with ADContextUser set to {}", contextUser.getUsername());
+        }
```

Comment:
Put this code to `NoarkFactory` instead to avoid duplicating for each and every case type.